File: CS/SLGridExample/SelectionStateToBrushConverter.cs
using System;
using System.Windows.Data;
using DevExpress.Xpf.Grid;
using System.Windows.Media;

namespace SLGridExample {
    public class SelectionStateToBrushConverter :IValueConverter {

        public SolidColorBrush NormalBrush { get; set; }
        public SolidColorBrush FocusedBrush { get; set; }
        public SolidColorBrush SelectedBrush { get; set; }
        public SolidColorBrush FocusedAndSelectedBrush { get; set; }

        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            SelectionState state = (SelectionState)value;

            if (state == SelectionState.Focused) {
                return FocusedBrush;
            }

            if (state == SelectionState.Selected) {
                return SelectedBrush;
            }

            if (state == SelectionState.FocusedAndSelected) {
                return FocusedAndSelectedBrush != null ? FocusedAndSelectedBrush : FocusedBrush;
            }

            return NormalBrush;
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}

File: VB/SLGridExample/SelectionStateToBrushConverter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Data
Imports DevExpress.Xpf.Grid
Imports System.Windows.Media

Namespace SLGridExample
	Public Class SelectionStateToBrushConverter
		Implements IValueConverter

		Public Property NormalBrush() As SolidColorBrush
		Public Property FocusedBrush() As SolidColorBrush
		Public Property SelectedBrush() As SolidColorBrush
		Public Property FocusedAndSelectedBrush() As SolidColorBrush

		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
			Dim state As SelectionState = CType(value, SelectionState)

			If state = SelectionState.Focused Then
				Return FocusedBrush
			End If

			If state = SelectionState.Selected Then
				Return SelectedBrush
			End If

			If state = SelectionState.FocusedAndSelected Then
				Return If(FocusedAndSelectedBrush IsNot Nothing, FocusedAndSelectedBrush, FocusedBrush)
			End If

			Return NormalBrush
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class
End Namespace
